### PR TITLE
[typescript-rxjs] add support for raw response and progressSubscriber

### PR DESCRIPTION
**Pull request by denyo**
This PR adds support for an optional parameter on controller functions in order to retrieve the status code or pass a `progressSubscriber` to receive events for longer lasting calls like file uploads.

It also makes use of the new `??` operator to support a relative `baseUrl` like `""`.

### PR checklist
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

**Comment by denyo** [comment]
@amakhrov thank you for your feedback.

The `raw` response approach definitely makes a lot of sense. 
Regarding `RequestOpts` and `AjaxRequest`, you are right. `RequestOpts` is basically a subset except for one or two props. I changed it to extend `AjaxRequest` with more explicit typing.

About the typescript version it is also brings the optional chaining operator `?`.
I guess most people use this generator for React or NodeJS projects. At least for React it is used since November in CRA.

Please have a look at the updated changes.

**Comment by macjohnny** [comment]
since this PR contains breaking-changes without fallback, we would have to wait for the 5.0.0 release (https://github.com/OpenAPITools/openapi-generator/milestone/20)

File: modules/openapi-generator/src/main/resources/typescript-rxjs/runtime.mustache [hunk 1]
```diff
@@ -35,19 +35,13 @@ export class Configuration {
     }
 
     get apiKey(): ((name: string) => string) | undefined {
-        const apiKey = this.configuration.apiKey;
-        if (!apiKey) {
-            return undefined;
-        }
-        return typeof apiKey === 'string' ? () => apiKey : apiKey;
+        const { apiKey } = this.configuration;
+        return apiKey ? (typeof apiKey === 'string' ? () => apiKey : apiKey) : undefined;
```
Comment (ermik) [inline modules/openapi-generator/src/main/resources/typescript-rxjs/runtime.mustache:39]: I don't think double ternary is appropriate there. It's obfuscating the conditional, it makes it less readable. I've addressed the nully coalescing in #5329, and I think webpack can take care of the compressing the code.

File: modules/openapi-generator/src/main/resources/typescript-rxjs/runtime.mustache [hunk 2]
```diff
@@ -35,19 +35,13 @@ export class Configuration {
...
     }
 
     get accessToken(): ((name: string, scopes?: string[]) => string) | undefined {
-        const accessToken = this.configuration.accessToken;
-        if (!accessToken) {
-            return undefined;
-        }
-        return typeof accessToken === 'string' ? () => accessToken : accessToken;
+        const { accessToken } = this.configuration;
+        return accessToken ? (typeof accessToken === 'string' ? () => accessToken : accessToken) : undefined;
```
Comment (ermik) [inline modules/openapi-generator/src/main/resources/typescript-rxjs/runtime.mustache:44]: Double ternary is problematic, as I described above.

**Comment by denyo** [comment]
I am thinking about adding support for a flag that will control weather or not the optional `progressSubscriber` will be part of the generated code.
What would be the best approach for this? Introducing an `--additional-properties` flag?
Any hints where I can see this? Or is looking at use of the `withInterfaces` flag enough even though it seems to be available in some generators but without any functionality like here.

**Comment by macjohnny** [comment]
@denyo can you please resolve the conflicts?

**Comment by ermik** [comment]
Do you think these changes are possible without losing readability? I've highlighted the issue with the double ternary above.

**Comment by denyo** [comment]
> Do you think these changes are possible without losing readability? I've highlighted the issue with the double ternary above.

I could improve the formatting but I am against having multiple exit points within a function since it makes it more difficult to debug is against the one entry point / one exit point rule.

**Comment by ermik** [comment]
`one entry point / one exit point rule` I don't think this applies here, since tenrary has two exit points (`true` & `false`) and the Nth ternary has 2ⁿ exits. The fact that all those exits happen on the same line of code, doesn't make it any easier to debug them, in fact, I would argue, it makes it harder.

@macjohnny I don't know if you have any strong opinions on this. Mine is certainly just that — an opinion — this shouldn't be held up over any sort of debate on best practices.

**Comment by macjohnny** [comment]
@ermik I agree with you, in my opinion, the double ternaries are less readable than two separate exit points, which, given that the resulting function is very small, are easy to spot.

moreover, I would prefer to avoid the `??` operator, as it breaks compatibility with older typescript versions without a real necessity to do so.

however, since this would be part of version 5.0.0, and users could still use 4.3.1 for older typescript versions, I think we should merge this one, as it adds features and improves the code, and @denyo has put a lot of effort in creating and improving the typescript-rxjs generator.

**Comment by ermik** [comment]
🎉  great job @denyo.

**Pull request merged**